File: r/secret-handshake/_iterations/1/secret-handshake.R
handshake <- function(n) {
  actions <- c(1, 2, 4, 8)
  names(actions) <- c("wink", "double blink", "close your eyes", "jump")
  output <- names(actions)[bitwAnd(n, actions) != 0]
  if (length(output) == 0) {
    c()
  } else if (bitwAnd(n, 16) != 0) {
    rev(output)
  } else {
    output
  }
}

File: clojure/secret-handshake/_iterations/1/src/secret_handshake.clj
(ns secret-handshake)

(defn commands [handshake]
  (cond-> []
          (pos? (bit-and handshake 2r1))     (conj "wink")
          (pos? (bit-and handshake 2r10))    (conj "double blink")
          (pos? (bit-and handshake 2r100))   (conj "close your eyes")
          (pos? (bit-and handshake 2r1000))  (conj "jump")
          (pos? (bit-and handshake 2r10000)) reverse))
